pkg/spoke/addon/configuration.go

```diff
@@ -12,30 +12,14 @@ import (
 	certificatesv1 "k8s.io/api/certificates/v1"
 )
 
-const (
-	defaultAddOnInstallationNamespace = "open-cluster-management-agent-addon"
-	installationNamespaceAnnotation   = "addon.open-cluster-management.io/installNamespace"
-	// registrationAnnotation includes the addOn registration configuration. If specified, it means the registration is enabled.
-	// The value is a json string, like
-	// `[ { "signerName":"signer1", "subject":{ "commonName": "addon1-agent", "organization":["addon1"], "organizationalUnit":["acm"] }]`
-	// where subject is optional.
-	registrationAnnotation = "addon.open-cluster-management.io/registrations"
-)
-
-// addonConfig contains addon configuration information.
-type addOnConfig struct {
-	AddOnName             string
-	InstallationNamespace string
-}
+const defaultAddOnInstallationNamespace = "open-cluster-management-agent-addon"
 
 // registrationConfig contains necessary information for addon registration
 // TODO: Refactor the code here once the registration configuration is available in spec of ManagedClusterAddOn
 type registrationConfig struct {
-	AddOnName             string `json:"addOnName,omitempty"`
-	InstallationNamespace string `json:"installNamespace,omitempty"`
-	// SignerName is used to create csr. The default value is "kubernetes.io/kube-apiserver-client".
-	SignerName string      `json:"signerName,omitempty"`
-	Subject    certSubject `json:"subject,omitempty"`
+	AddOnName             string                           `json:"addOnName,omitempty"`
```

Comment:
Do not need json part, and it does not need to be public

Reply:
updated, using registration hash instead of registrationConfig